### PR TITLE
Making bazel build working

### PR DESCRIPTION
**Pull request by donnadionne**

### CODE REVIEW

**Review by jtattermusch — COMMENTED** [review]
Ok, yet another dependency being added.

As always, make sure you comply with https://github.com/grpc/grpc/tree/master/third_party#third-party-libraries and please be prepared to help with the build maintenance and also be ready to step in to address followup issues (there's always some) after this is added. I'm explicitly mentioning that because too often folks add a dependency and then they hope someone else with magically find time to take care of all the followup maintenance, responding to user issues and fixing build issues.

Also, please provide a reasoning for why this dependency is needed.

Let me know once this is fully ready to review.

File: .gitmodules [hunk 1]
```diff
@@ -46,3 +46,6 @@
 [submodule "third_party/libuv"]
 	path = third_party/libuv
 	url = https://github.com/libuv/libuv.git
+[submodule "third_party/xxhash"]
+	path = third_party/xxhash
+	url = git://github.com/Cyan4973/xxHash.git
```
[the remark this answers is not in the document]
Comment (donnadionne) [reply .gitmodules]: fixed

**Comment by donnadionne** [comment]
Hi!  I am ready for a review on this change.

Reason for importing xxhash are the following:
1. this is a fast and light library
2. google3 already uses it
3. to be consistent with envoy implementation of the same feature: ring hash
4. without it we would have to implement our own hashing library

**Comment by donnadionne** [comment]
I do still have about 5 build failures, would appreciate any pointers on how I can fix them.
1. Artifacts MacOS: Dispatch only on x85 and x86_64
2. ObjC example: complaint of a duplicated task
3. ObjC iOS: some commented code are being compiled and treated as error
4. Distribution windows: Distribution Linux already pass, not sure what additional change is needed for windows
5. ios size: this one I think has been failing prior to this as well

I spoke to Nico and one of the suggestion is to inline the implementation and prevent the building of the library, I am not too sure what I need to do to turn off the building of the library as I think we still want to import it as a submodule which seems to get built.

**Comment by donnadionne** [comment]
going with a simpler solution

**Pull request closed without merging**